hal/halx86/acpi/madt.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Source File for MADT parses 
+ * PROGRAMMERS: Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include <hal.h>
+#include <acpi.h>
+/* ACPI_BIOS_ERROR defined in acoutput.h and bugcodes.h */
+#undef ACPI_BIOS_ERROR
+#include <smp.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS ********************************************************************/
+
+PROCESSOR_IDENTITY HalpStaticProcessorIdentity[MAX_CPUS] = {{0}};
+PPROCESSOR_IDENTITY HalpProcessorIdentity = NULL;
+HALP_APIC_INFO_TABLE HalpApicInfoTable;
+UINT32 PhysicalProcessorCount = 0;
+ACPI_TABLE_MADT *MadtTable;
+ACPI_SUBTABLE_HEADER *AcpiHeader;
+ACPI_MADT_LOCAL_APIC *LocalApic;
+
+/* FUNCTIONS ******************************************************************/
+
+VOID
+HalpParseApicTables(IN PLOADER_PARAMETER_BLOCK LoaderBlock)
```

Comment:
Maybe you could take the opportunity to use the new SAL annotations:
```suggestion
HalpParseApicTables(
    _In_ PLOADER_PARAMETER_BLOCK LoaderBlock)
```